pkg/analyze/velero.go

```diff
@@ -0,0 +1,631 @@
+package analyzer
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+const (
+	DefaultVeleroNamespace = "velero"
+)
+
+type AnalyzeVelero struct {
+	analyzer *troubleshootv1beta2.VeleroAnalyze
+}
+
+func (a *AnalyzeVelero) Title() string {
+	title := a.analyzer.CheckName
+	if title == "" {
+		title = "Velero"
+	}
+
+	return title
+}
+
+func (a *AnalyzeVelero) IsExcluded() (bool, error) {
+	return isExcluded(a.analyzer.Exclude)
+}
+
+func (a *AnalyzeVelero) Analyze(getFile getCollectedFileContents, findFiles getChildCollectedFileContents) ([]*AnalyzeResult, error) {
+	results, err := a.veleroStatus(a.analyzer, getFile, findFiles)
+	if err != nil {
+		return nil, err
+	}
+	for i := range results {
+		results[i].Strict = a.analyzer.Strict.BoolOrDefaultFalse()
+	}
+	return results, nil
+}
+
+func (a *AnalyzeVelero) veleroStatus(analyzer *troubleshootv1beta2.VeleroAnalyze, getFileContents getCollectedFileContents, findFiles getChildCollectedFileContents) ([]*AnalyzeResult, error) {
+	ns := DefaultVeleroNamespace
+	if analyzer.Namespace != "" {
+		ns = analyzer.Namespace
+	}
+
+	excludeFiles := []string{}
+
+	// get backuprepositories.velero.io
+	backupRepositoriesDir := GetVeleroBackupRepositoriesDirectory(ns)
+	backupRepositoriesGlob := filepath.Join(backupRepositoriesDir, "*.json")
```

Comment:
I do not think we respect the namespace in this analyser. Its not used at all. Should we either use it, or remove it totally? I'm for removing it. I do not see a usecase where we would want to pick a namespace to filter analysis.

Reply:
Yea no need after removing collector code. 👍🏻

Reply:
actually I forgot, we're still using namespace when picking up symlinked logs

Reply:
`name: velero/logs` - that should suffice. I'll remove it entirely.